### PR TITLE
user: password validator

### PR DESCRIPTION
**Pull request by Garfield-fr**
The password is mandatory if the user is not loaded,
otherwise it is optional

* Adds validator on new password on change password dialog.
* Fixes the identifier of the delete button so that it is unique.

Co-Authored-by: Bertrand Zuchuat <bertrand.zuchuat@rero.ch>

## Code review check list

- [ ] Commit message template compliance.
- [ ] Commit message without typos.
- [ ] File names.
- [ ] Functions names.
- [ ] Functions docstrings.
- [ ] Unnecessary commited files?
- [ ] Cypress tests successful?

### CODE REVIEW

File: projects/public-search/src/app/patron-profile/patron-profile-password/patron-profile-password.component.ts [hunk 1]
```diff
@@ -159,6 +165,27 @@ export class PatronProfilePasswordComponent {
...
+  /** Async validator for the password validator */
+  validatePassword(): any {
+    return {
+      expression: (control: UntypedFormControl) => {
+        const value = control.value;
+        if (value == null || value.length === 0) {
+          return of(true);
+        }
+        return this._userApiService.validatePassword(value).pipe(
+          debounceTime(1000),
```
Comment (zannkukai) [inline projects/public-search/src/app/patron-profile/patron-profile-password/patron-profile-password.component.ts:177]: 1s seems maybe a little bit excessive ? but certainly not critic

**Review by zannkukai — APPROVED** [review]

File: projects/admin/src/app/record/custom-editor/user-id-editor/user-id-editor.component.html [hunk 2]
```diff
@@ -24,12 +24,12 @@ <h4 id="dialog-sizes-name2" class="modal-title pull-left" translate>Personal inf
           (search)="searchValueUpdated($event)" class="form-inline mr-3" [focus]="false" [displayLabel]="false">
         </ng-core-search-input>
       </form>
-      <button type="button" id="editor-delete-button" class="btn btn-outline-danger btn-sm mr-1"
+      <button type="button" id="user-editor-delete-button" class="btn btn-outline-danger btn-sm mr-1"
```
Comment (jma) [inline projects/admin/src/app/record/custom-editor/user-id-editor/user-id-editor.component.html:27]: Should be mentioned in the commit message.

File: projects/public-search/src/app/api/user-api.service.ts [hunk 3]
```diff
@@ -37,6 +37,10 @@ export class UserApiService {
   updatePassword(data: IPassword): Observable<any> {
     return this._httpClient.post('/api/change-password', data);
   }
+
+  validatePassword(password: string): Observable<any> {
+    return this._httpClient.post('/api/user/password/validate', {'password': password});
```
Comment (jma) [inline projects/public-search/src/app/api/user-api.service.ts:42]: You should use: `getEndpointByType` or similar.

**Review by jma — APPROVED** [review]

File: projects/public-search/src/app/patron-profile/patron-profile-password/patron-profile-password.component.ts [hunk 4]
```diff
@@ -1,6 +1,6 @@
 /*
  * RERO ILS UI
- * Copyright (C) 2022 RERO
+ * Copyright (C) 2022-2023 RERO
```
[the remark this answers is not in the document]
Comment (Garfield-fr) [reply projects/public-search/src/app/patron-profile/patron-profile-password/patron-profile-password.component.ts:3]: The first date is the date of creation of the file and the second date changes if the file is modified.

File: projects/public-search/src/app/patron-profile/patron-profile-password/patron-profile-password.component.ts [hunk 5]
```diff
@@ -81,9 +81,12 @@ export class PatronProfilePasswordComponent {
           type: 'password',
           label: 'New password',
           required: true,
-          minLength: 6,
+          minLength: 8,
```
[the remark this answers is not in the document]
Comment (Garfield-fr) [reply projects/public-search/src/app/patron-profile/patron-profile-password/patron-profile-password.component.ts:84]: This saves many requests to the server 😉

**Pull request merged**